v1/plugins/logs/plugin.go

```diff
@@ -919,21 +952,83 @@ func (p *Plugin) reconfigure(config interface{}) {
 
 	p.logger.Info("Decision log uploader configuration changed.")
 	p.config = *newConfig
+
+	switch newConfig.Reporting.BufferType {
+	case eventBufferType:
+		if p.runningBuffer == sizeBufferType {
+			p.flushDecisions(ctx)
+		}
+
+		if p.eventBuffer == nil {
+			p.eventBuffer = newEventBuffer(
+				*p.config.Reporting.BufferSizeLimitEvents,
+				p.manager.Client(p.config.Service),
+				*p.config.Resource,
+				*p.config.Reporting.UploadSizeLimitBytes)
+		} else {
+			dropped, errs := p.eventBuffer.Reconfigure(
+				*p.config.Reporting.BufferSizeLimitEvents,
+				p.manager.Client(p.config.Service),
+				*p.config.Resource,
+				*p.config.Reporting.UploadSizeLimitBytes)
+
+			for _, err := range errs {
+				if err == nil {
+					continue
+				}
+				if errors.Is(err, droppedNDCache{}) {
+					p.incrMetric(logNDBDropCounterName)
+				} else {
+					p.incrMetric(logEncodingFailureCounterName)
+				}
+				p.logger.Error("Log encoding failed: %v.", err)
+			}
+			for range dropped {
+				p.incrMetric(logBufferEventLimitExDropCounterName)
+			}
+		}
+
+		p.runningBuffer = eventBufferType
+	case sizeBufferType:
+		if p.runningBuffer == eventBufferType {
+			_, err := p.eventBuffer.Upload(ctx)
+			if err != nil {
+				p.setStatus(err)
```

Comment:
We don't need to set the status when flushing the `size` buffer above? Or maybe that happens a bit further down there?

Reply:
I think the status should be set, `flushDecisions` was only being used when the plugin stopped so the error was only logged. `flushDecisions` might not be what we want here anyway because on upload failure it loops until the context is cancelled. I think it should just try uploading the buffer once and then continue. Updated it to call `oneShot` instead.